Google.Api.Gax.Grpc/ApiCallLoggingExtensions.cs

```diff
@@ -0,0 +1,87 @@
+﻿/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Grpc.Core;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax.Grpc
+{
+    internal static class ApiCallLoggingExtensions
+    {
+        // By design, the code is mostly duplicated between the async and sync calls.
+
+        #region Unary calls
+        // Async call
+        internal static Func<TRequest, CallSettings, Task<TResponse>> WithLogging<TRequest, TResponse>(
+            this Func<TRequest, CallSettings, Task<TResponse>> fn, ILogger logger) =>
+            async (request, callSettings) =>
+            {
+                logger?.LogTrace("Starting asynchronous API call.");
```

Comment:
Not sure if we should add an EventID here or if we should let loggers handle that?

Reply:
Loggers will add event IDs, I believe.